app/controllers/concerns/report_embeddable.rb

```diff
@@ -32,7 +33,10 @@ def build_report_data(options = {})
   # returns true if no errors, false otherwise
   def run_and_handle_errors
     begin
-      @report.run(current_ability)
+      Rails.cache.fetch(@report.cache_key_with_responses(current_mission)) do
+        @report.run(current_ability)
+        @report
+      end
```

Comment:
@hooverlunch The problem is that `@report.run` doesn't return the `@report`. So what we would be caching instead is the return of @report.run if we removed the last line on the block.

I don't like the way it is right now too. If we could change the return of all `run` methods to return `self`, we could do the way you suggested, but i'm not sure if this would affect other parts of the app. It looks like (from a quick search) that the return of run is not being used, but you must know better.

Let me know what you think.

Thanks!